client/kafka/builder.go

```diff
@@ -123,33 +121,57 @@ func (ab *AsyncBuilder) WithEncoder(enc encoding.EncodeFunc, contentType string)
 	return ab
 }
 
-// Create constructs the AsyncProducer component by applying the gathered properties.
-func (ab *AsyncBuilder) Create() (*AsyncProducer, error) {
+// WithSync determines whether the producer is synchronous or asynchronous; default is asynchronous.
+func (ab *Builder) WithSync(sync bool) *Builder {
+	ab.sync = sync
+
+	log.Info(fieldSetMsg, "required sync", sync)
+
+	return ab
+}
+
+// Create constructs the Producer/AsyncProducer component by applying the gathered properties.
+func (ab *Builder) Create() (*KafkaProducer, error) {
```

Comment:
A good idea is to create different methods for creating sync and async e.g. CreateSync() and CreateAsync().
This way we can also return other arguments like error channels for async and also make the code look cleaner without the need for the if switches.

Reply:
Good idea, then I can remove `Error() <-chan error` from the `Producer` interface and return it from the `CreateAsync()` method, making it immediately visible for developers that they need to handle it and at the same time preserving the same `Producer` interface for both the sync/async versions.